### PR TITLE
Remove non-printable characters from URL before redirecting

### PR DESCRIPTION
**Pull request by andrewleith**
# Summary | Résumé
Remove non-printable characters from URL before redirecting inside of `/set-lang`

### CODE REVIEW

File: tests/app/test_utils.py [hunk 1]
```diff
@@ -694,6 +695,23 @@ def test_set_lang_bad_route(
     assert page.select_one("h1").text.strip() == "Page could not be found"
 
 
+def test_set_lang_bad_route_with_control_characters(
```
Comment (jimleroyer) [inline tests/app/test_utils.py:698]: Do we have regression tests that cover valid routes that should not be blocked?

File: app/main/views/set_lang.py [hunk 2]
```diff
@@ -20,4 +20,7 @@ def set_lang():
         # redirect to main page in case of an invalid redirect
         url = url_for("main.show_accounts_or_dashboard")
 
+    # remove non-printable characters from url
+    url = "".join(ch for ch in url if ch.isprintable())
```
Comment (jimleroyer) [inline app/main/views/set_lang.py:24]: This feels like a sanitization library exists for this, but that should be enough.

**Review by jimleroyer — APPROVED** [review]
I left a few comments but it seems good to avoid the bug immediately.

File: app/main/views/set_lang.py (hunk 2 above)
Comment (andrewleith) [reply app/main/views/set_lang.py:24]: Yeah I thought the same thing.  I looked briefly and found Bleach and Schema but neither seemed to work for what we need here.  I prioritized doing it quickly vs. looking in depth though, so there likely is something out there.

File: tests/app/test_utils.py (hunk 1 above)
Comment (andrewleith) [reply tests/app/test_utils.py:698]: No - but if we have an existing route with non-printable characters, that would throw an error anyway.  I can add a todo to make a more thorough test here for existing routes though.

**Pull request merged**